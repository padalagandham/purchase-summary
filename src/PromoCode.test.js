import React from 'react';
import ReactDOM from 'react-dom';
import Promocode from './PromoCode';
import Adapter from 'enzyme-adapter-react-16';
 import { shallow, configure } from 'enzyme';

describe('Promo Component', () => {
    const div = document.createElement('div');
    configure({adapter: new Adapter()});
    
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Promocode />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('promoFormContainer should exist', () => {
        const component = shallow(<Promocode debug />);
        expect(component).toMatchSnapshot();
    });

    it('should have toggleContainer  to show and hide', () => {
        const component = shallow(<Promocode />);
        expect(component.find('.toggleContainer').length).toEqual(1)
        component.unmount();
    });

    it('click on toggleContainer to show promo code form', () => {
        const component = shallow(<Promocode />);
        component.find('.toggleContainer').simulate("click");
        expect(component.find('.promoFormContainer').length).toEqual(1)
        component.unmount();
    });

});