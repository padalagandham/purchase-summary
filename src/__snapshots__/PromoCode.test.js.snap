// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Promo Component promoFormContainer should exist 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Promocode
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="toggleContainer"
          onClick={[Function]}
        >
          <div
            className="label"
          >
            Apply promo code
          </div>
          <div
            aria-hidden="true"
            className="plus icon"
          />
        </div>,
        "",
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="label"
            >
              Apply promo code
            </div>,
            <div
              aria-hidden="true"
              className="plus icon"
            />,
          ],
          "className": "toggleContainer",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Apply promo code",
              "className": "label",
            },
            "ref": null,
            "rendered": "Apply promo code",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-hidden": "true",
              "className": "plus icon",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      "",
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="toggleContainer"
            onClick={[Function]}
          >
            <div
              className="label"
            >
              Apply promo code
            </div>
            <div
              aria-hidden="true"
              className="plus icon"
            />
          </div>,
          "",
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="label"
              >
                Apply promo code
              </div>,
              <div
                aria-hidden="true"
                className="plus icon"
              />,
            ],
            "className": "toggleContainer",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Apply promo code",
                "className": "label",
              },
              "ref": null,
              "rendered": "Apply promo code",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-hidden": "true",
                "className": "plus icon",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        "",
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
